### PR TITLE
Validation rules for providers path urls

### PR DESCRIPTION
**Pull request by dsgouda**
@olydis @amarzavery PTAL

### CODE REVIEW

**Comment by olydis** [comment]
Have you tested this rule against the specs repo? Playing around with a few paths I thing that this rule might be too strict, i.e. complains on paths that (I think) are actually fine:
image
(note: the first two paths do not match your regex, while the last two do.)

File: src/modeler/AutoRest.Swagger/Validation/ProvidersPathValidation.cs [hunk 1]
```diff
@@ -13,10 +13,10 @@
 
 namespace AutoRest.Swagger.Validation 
 {
-    public class ProviderPathsValidation : TypedRule<Dictionary<string, Dictionary<string, Operation>>>
+    public class ProvidersPathValidation : TypedRule<Dictionary<string, Dictionary<string, Operation>>>
     {
-        private readonly Regex FullRegex = new Regex(@"\/Subscriptions\/\{.+\}\/ResourceGroups\/\{.+\}\/providers\/[^\/]+(\/[^\/]+\/\{[^\}]+\})*(\/[\/^])?([\?]?[^\/]+)*$", RegexOptions.IgnoreCase);
-        private readonly Regex ProviderRegex = new Regex(@"Subscriptions\/\{.+\}\/ResourceGroups\/\{.+\}\/providers\/.+$", RegexOptions.IgnoreCase);
+        private Regex FullRegex = new Regex(@"\/Subscriptions\/\{.+\}\/ResourceGroups\/\{.+\}\/providers\/[^\/]+(\/[^\/]+\/\{[^\}]+\})*(\/[^\/]+)?([\?]?[^\/]+)*$", RegexOptions.IgnoreCase);
```
Comment (olydis) [inline src/modeler/AutoRest.Swagger/Validation/ProvidersPathValidation.cs]: why did you remove the `readonly`?

Comment (olydis) [inline src/modeler/AutoRest.Swagger/Validation/ProvidersPathValidation.cs]: regex should be equivalent to `\/Subscriptions\/\{.+\}\/ResourceGroups\/\{.+\}\/providers\/[^\/]+(\/[^\/]+\/\{[^\}]+\})*(\/[^\/]+)?$` (shorter)

Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/ProvidersPathValidation.cs]: I'm trying to catch the pattern that x-ms-paths follow (i.e. /path?foo={fooid})
I know it's a little overboard, but don't see the harm

Comment (olydis) [inline src/modeler/AutoRest.Swagger/Validation/ProvidersPathValidation.cs]: thought you wanted a simpler regex earlier ;-) also note that your regex also matches `.../{foo}bar` so it does not really catch the pattern! Note also that your test case is failing on Jenkins

File: src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs [hunk 2]
```diff
@@ -427,6 +435,17 @@ public void LongRunningResponseDefined()
...
             messages.AssertOnlyValidationMessage(typeof(LongRunningResponseValidation), 0);
         }
+
+        /// <summary>
+        /// Verifies that a providers path is of proper format
+        /// </summary>
+        [Fact]
+        public void ProvidersPathValidJson()
+        {
+            var messages = ValidateSwagger(Path.Combine("Resource", "Swagger", "Validation", "positive", "providers-path-valid.json"));
```
Comment (olydis) [inline src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs]: `Core.Utilities.Extensions.CodeBaseDirectory, ` missing

Comment (dsgouda) [inline src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs]: I understand I am being inconsistent here but not all paths in this file have ```Core.Utilities.Extensions.CodeBaseDirectory``` just wondering why that is so
